File: OOPSeminar5/python_mvc_app/controller/controller_class.py
from typing import List
from controller.command import Command
from controller.interfaces.i_get_model import iGetModel
from controller.interfaces.i_get_view import iGetView
from model.core.student import Student


class ControllerClass:
    def __init__(self, model: iGetModel, view: iGetView):
        self.model = model
        self.view = view
        self.students = []


    def testData(self, studs: List['Student']) -> bool:
        return len(studs) > 0

    def update(self) -> None:
        #MVP
        self.students = self.model.getStudents()

        if self.testData(self.students):
            self.view.printAllStudent(self.students)
        else:
            print("Список студентов пуст!")
            
        #MVC
        #self.view.printAllStudent(self.model.getStudents())

    def run(self) -> None:
        com = Command.NONE
        getNewIter = True

        while getNewIter:
            command = self.view.prompt("Введите команду:")
            com = Command[command.upper()]

            if com == Command.EXIT:
                getNewIter = False
                print("Выход из программы")
            elif com == Command.LIST:
                self.view.printAllStudent(self.model.getStudents())

File: OOPSeminar5/MVCExample/MVCStudentApp/src/Controller/ControllerClass.java
package Controller;

import java.util.ArrayList;
import java.util.List;

import Controller.Interfaces.iGetModel;
import Controller.Interfaces.iGetView;
import Model.ModelClassList;
import Model.Core.Student;
import View.ViewClass;

public class ControllerClass {

    private iGetModel model;
    private iGetView view;
    private List<Student> studentBuffer = new ArrayList<>();

    public ControllerClass(iGetModel model, iGetView view) {
        this.model = model;
        this.view = view;
    }

    private boolean testData(List<Student> stud)
    {
        if(stud.size()>0)
        {
            return true;
        }
        else
        {
            return false;
        }
    }

    public void update()
    {

        //MVP
        studentBuffer = model.getStudents();

        if(testData(studentBuffer))
        {
            view.printAllStudents(studentBuffer);
        }
        else
        {
            System.out.println("Список студентов пуст!");
        }


        //MVC
        //view.printAllStudents(model.getStudents());
    } 
    
    public void run()
    {
        Command com = (Command)Command.NONE;
        boolean getNewIter = true;
        while(getNewIter)
        {
            String command = view.prompt("Введите команду:");
            com = Command.valueOf(command.toUpperCase());
            switch(com)
            {
                case EXIT:
                   getNewIter = false;
                   System.out.println("Выход из программы");
                   break;
                case LIST:
                   view.printAllStudents(model.getStudents());
                   break;
            }
        }
    }
}
